packages/stable-sdk/src/methods/executeRoute/index.ts

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2025 Stable Technologies Inc
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+import { ViemEvmClient } from "@stable-io/cctp-sdk-viem";
+import type { Network } from "@stable-io/cctp-sdk-definitions";
+import { avaxRouterContractAddress } from "@stable-io/cctp-sdk-cctpr-definitions";
+import { Route, SDK } from "../../types/index.js";
+
+import { executeRouteSteps } from "./executeRouteSteps.js";
+import { CctpAttestation, findTransferAttestation } from "./findTransferAttestation.js";
+import { findTransferRedeem } from "./findTransferRedeem.js";
+import { Redeem } from "src/types/redeem.js";
+
+export type ExecuteRouteDeps<N extends Network> = Pick<SDK<N>, "getNetwork" | "getRpcUrl" | "getSigner">;
+
+export const $executeRoute =
+  <N extends Network>({
+    getSigner,
+    getNetwork,
+    getRpcUrl,
+  }: ExecuteRouteDeps<N>): SDK<N>["executeRoute"] =>
+  async (route: Route) => {
+    const signer = getSigner(route.intent.sourceChain);
+    const network = getNetwork();
+    const rpcUrl = getRpcUrl(route.intent.sourceChain);
+    const client = ViemEvmClient.fromNetworkAndDomain(
+      network,
+      route.intent.sourceChain,
+      rpcUrl,
+    );
+
+    /**
+     * @todo: we could get and return signatures along with the transactions and
+     *        we'd be providing all infromation the integrator may possibly need.
+     * @todo: review return value of executeRouteSteps. Having only the tx hashes
+     *        makes it hard/unreliable to know which is the transfer transaction.
+     *        For the time being we can get our way by getting the last tx, but
+     *        this wouldn't resist integrating protocols with multiple transactions.
+     */
+    const transactions = await executeRouteSteps(network, route, signer, client);
+    const transferTx = transactions.at(-1)!; // there's always 1 or 2 hashes.
+
+    const attestations = [] as CctpAttestation[];
+    const redeems = [] as Redeem[];
```

Comment:
This achieves the same thing and is safer.
```suggestion
    const attestations: CctpAttestation[] = [];
    const redeems: Redeem[] = [];
```